### PR TITLE
Update generate scripts to use latest code_gen functions

### PR DESCRIPTION
**Pull request by ArangoGutierrez**

### CODE REVIEW

**Comment by ArangoGutierrez** [comment]
Ha!  doesn't need rebase :) 
@marquiz PTAL

File: hack/update_codegen.sh [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  --output-base "${NFD_ROOT}" \
+  --boilerplate ${NFD_ROOT}/hack/boilerplate.go.txt
+
+kube::codegen::gen_client \
+  --input-pkg-root sigs.k8s.io/node-feature-discovery/pkg/apis \
+  --output-pkg-root sigs.k8s.io/node-feature-discovery/pkg/generated \
+  --output-base "${NFD_ROOT}" \
+  --boilerplate ${NFD_ROOT}/hack/boilerplate.go.txt \
+  --with-watch \
+  --with-applyconfig
```
Comment (marquiz) [inline hack/update_codegen.sh:62]: Do we need/want this?

**Review by marquiz — COMMENTED** [review]
Thanks @ArangoGutierrez.

Some notes below. Plus the PR needs a rebase.

Also, the commit message is misleading (and short). It's not updating the codegen tools but updating/revamping the generator script.

**Comment by marquiz** [comment]
> Ha! doesn't need rebase :)

OK, so it seems. However, could you rebase so that it becomes cleaerer what this PR actually changes

File: Dockerfile_generator [hunk 2]
```diff
@@ -21,3 +21,7 @@ RUN curl -LO https://github.com/protocolbuffers/protobuf/releases/download/v25.3
 
 # Expect to be working with nfd
 WORKDIR /go/node-feature-discovery
+
+# We need to set the /go/node-feature-discovery directory as a safe directory.
+# This allows git commands to run in the container.
```
Comment (marquiz) [inline Dockerfile_generator:26]: Maybe add a note that this is only for MacOs?

File: pkg/apis/nfd/v1alpha1/types.go [hunk 3]
```diff
@@ -31,11 +31,12 @@ type NodeFeatureList struct {
 	Items []NodeFeature `json:"items"`
 }
 
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
```
Comment (marquiz) [inline pkg/apis/nfd/v1alpha1/types.go:35]: Why were these moved? If they need to be moved then the kubebuilder stuff, too?

File: pkg/apis/nfd/v1alpha1/types.go [hunk 4]
```diff
@@ -111,13 +112,14 @@ type NodeFeatureRuleList struct {
 	Items []NodeFeatureRule `json:"items"`
 }
 
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
```
Comment (marquiz) [inline pkg/apis/nfd/v1alpha1/types.go:117]: Same here, do we need to move these?

File: pkg/generated/applyconfiguration/internal/internal.go
Comment (marquiz) [inline pkg/generated/applyconfiguration/internal/internal.go:1]: I think all the applyconfiguration files should be dropped from the PR now that `--with-applyconfig` was dropped from the codegen args. Right?

**Review by marquiz — COMMENTED** [review]
Looks better and more confined, thanks for the update. A few more comments below. 

Also, please update the commit message (https://github.com/kubernetes-sigs/node-feature-discovery/pull/1605#pullrequestreview-1927344426)

**Comment by ArangoGutierrez** [comment]
/retitle Update generate scripts to use latest code_gen functions

**Review by marquiz — APPROVED** [review]
/lgtm
